### PR TITLE
Modified Signature

### PR DESCRIPTION
**Pull request by dawidsula26**

### CODE REVIEW

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala [hunk 1]
```diff
@@ -16,23 +16,18 @@ class GenericFunctionError(messageInner: String) extends ExpressionParseError {
...
-class NoVarArgSignature(val name: String, val types: List[TypingResult]) extends Signature {
-  def display = s"$name(${typesToString(types)})"
+  def display: String = varArg match {
+    case Some(x) => s"$name(${typesToString(noVarArgs :+ x)}...)"
+    case None => s"$name(${typesToString(noVarArgs)})"
+  }
 
   override def equals(obj: Any): Boolean = obj match {
-    case x: NoVarArgSignature => name == x.name && types == x.types
+    case x: Signature => name == x.name && noVarArgs == x.noVarArgs && varArg == x.varArg
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala:29]: Why it is not just a case class?

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala [hunk 2]
```diff
@@ -16,23 +16,18 @@ class GenericFunctionError(messageInner: String) extends ExpressionParseError {
 }
 
 
-sealed abstract class Signature {
-  def display: String
-
-  protected def typesToString(types: List[TypingResult]): String =
+final class Signature(val name: String, val noVarArgs: List[TypingResult], val varArg: Option[TypingResult]) {
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala:19]: Can't we do the same with `VarArgsMethodInfo` and `NoVarArgsMethodInfo`?

Comment (mproch) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala:19]: :+1:

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala (hunk 1 above)
Comment (mproch) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala:29]: :+1:

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala (hunk 2 above)
Comment (dawidsula26) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/generics/ExpressionParseError.scala:19]: I have already done it, but its with adding support for varArgs in generic functions

**Review by arkadius — APPROVED** [review]

**Pull request merged**